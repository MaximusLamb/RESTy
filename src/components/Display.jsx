import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ url,  body, onSubmit, onChange }) => (
  <form onSubmit={onSubmit}>
    <label>
      URL
      <input type='text' value={url} onChange={onChange}/>
    </label>
    <br></br>
    <section>
      <br></br>
      <label>
        GET
        <input type='radio' value='get' name='method'></input>
      </label>
      <br></br>
      <label>
        POST
        <input type='radio' value='post' name='method'></input>
      </label>
      <br></br>
      <label>
        PUT
        <input type='radio' value='put' name='method'></input>
      </label>
      <br></br>
      <label>
        DELETE
        <input type='radio' value='delete' name='method'></input>
      </label>
      <br></br>
      <label>Input</label>
      <textarea type='text' value={body} name='input'></textarea>
    </section>
    <button>Submit</button>
  </form>
);

Display.propTypes = {
  url: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequred,
  onChange: PropTypes.func.isRequired
};

export default Display;
