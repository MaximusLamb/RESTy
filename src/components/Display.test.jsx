import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders the Display', () => {
    const wrapper = shallow(<Display url="butt@butt.com" body="fun things time"  onChange={() => {}} onSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
