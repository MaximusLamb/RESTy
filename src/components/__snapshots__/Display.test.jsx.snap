// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders the Display 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    URL
    <input
      onChange={[Function]}
      type="text"
      value="butt@butt.com"
    />
  </label>
  <br />
  <section>
    <br />
    <label>
      GET
      <input
        name="method"
        type="radio"
        value="get"
      />
    </label>
    <br />
    <label>
      POST
      <input
        name="method"
        type="radio"
        value="post"
      />
    </label>
    <br />
    <label>
      PUT
      <input
        name="method"
        type="radio"
        value="put"
      />
    </label>
    <br />
    <label>
      DELETE
      <input
        name="method"
        type="radio"
        value="delete"
      />
    </label>
    <br />
    <label>
      Input
    </label>
    <textarea
      name="input"
      type="text"
      value="fun things time"
    />
  </section>
  <button>
    Submit
  </button>
</form>
`;
